app/controllers/pieces_controller.rb

```diff
@@ -0,0 +1,21 @@
+class PiecesController < ApplicationController
+  
+  def update
+    if selected_piece.valid_move?(params[:piece][:x_position].to_i,params[:piece][:y_position].to_i)
+      selected_piece.update_attributes(piece_params)
+      render json: { response: "Valid Move"}
+    else
+      render json: { response: "Invalid Move" }, status: :unprocessable_entity
```

Comment:
Pinging off the above, with errors sometimes it does make sense to include some error text. It's best to encapsulate though  `response: {error: 'Invalid Move'}`